cmd/apidocs/update.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apidocs
+
+import (
+	"fmt"
+	"strconv"
+
+	"internal/apiclient"
+	"internal/client/apidocs"
+
+	"github.com/spf13/cobra"
+)
+
+// UpdateCmd to create a catalog items
+var UpdateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update an existing catalog item",
+	Long:  "Update an existing catalog item",
+	Args: func(cmd *cobra.Command, args []string) (err error) {
+		_, err = strconv.ParseBool(published)
+		if err != nil {
+			return fmt.Errorf("published must be a boolean value: %v", err)
+		}
+
+		_, err = strconv.ParseBool(anonAllowed)
+		if err != nil {
+			return fmt.Errorf("allow-anon must be a boolean value: %v", err)
+		}
+
+		_, err = strconv.ParseBool(requireCallbackUrl)
+		if err != nil {
+			return fmt.Errorf("require-callback-url must be a boolean value: %v", err)
+		}
+
+		return apiclient.SetApigeeOrg(org)
+	},
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		_, err = apidocs.Update(siteid, name, title, description, published,
+			anonAllowed, apiProductName, requireCallbackUrl, imageUrl, categoryIds)
+		return
+	},
+}
+
+func init() {
+	UpdateCmd.Flags().StringVarP(&name, "name", "n",
+		"", "Catalog name")
+	UpdateCmd.Flags().StringVarP(&title, "title", "t",
+		"", "The user-facing name of the catalog item")
```

Comment:
gorun apidocs update --help
panic: unable to redefine 't' shorthand in "update" flagset: it's already used for "title" flag